### PR TITLE
Fix “Disable vehicle limits” cheat giving all new rides 32 trains with 12 cars

### PR DESCRIPTION
**Pull request by Basssiiie**
Hey all,

This small fix should fix problems like #16026 where using the "disable vehicle limits" cheat would give all newly created rides 32 trains with sometimes also 12 cars per train, whether it was a flatride like merry-go-round or one car tracked rides like log flumes. Another added bonus is that all 32 merry-go-round trains are valid vehicles, giving you cute merry-go-round a hidden capacity of 16 x 32 = 512 seats! (That's one busy merry-go-round, I wouldn't want it to break down)

I could not find any other issues related to this issue in the issue tracker, so if you know any, please mention them.

## The fix
The fix changes the initialization values for `proposed_num_vehicles` and `proposed_num_cars_per_train` to more reasonable values when the cheat is enabled. Usually, without the cheat, these base values are immediately capped by the allowed limits. With the cheat, these limits are skipped (duh). Since the cheat skips the limits anyway, it seems better to initialize them with smaller values.

- For flatrides the amount of trains is set equal to the default amount of flatride vehicles, e.g. 1 for ferriswheel/freefall/lift, 4 for space rings, 12 for dodgems etc.
- For other rides, I did not see any relevant value in `rideEntry` that could be useful to determine a fitting proposed train amount. I set it to 12 as a kind of randomly picked value that is closer to what people would realistically pick in most cases on rides like log flumes, car rides. For rides like looping coasters it isn't that far off from a realistic number for those.
- The initial amount of vehicles per train is set to the maximum allowed vehicles per train without the cheat, e.g. dingy boats are 1, looping coaster trains are 8.

This fix only changes the initialization values. The actual minimum and maximum possible trains and vehicles are unaffected and unchanged.

I also incremented the network version, because I think it's probably necessary. But if it isn't, then please let me know.

I hope it helps and would love to hear it if there is any feedback. 😃 Thank you for your time!

### CODE REVIEW

File: src/openrct2/actions/RideCreateAction.cpp [hunk 1]
```diff
@@ -154,10 +154,25 @@ GameActions::Result RideCreateAction::Execute() const
     ride->vehicle_change_timeout = 0;
     ride->num_stations = 0;
     ride->num_vehicles = 1;
-    ride->proposed_num_vehicles = 32;
+    if (gCheatsDisableTrainLengthLimit)
+    {
+        // Reduce amount of proposed trains to prevent 32 trains from always spawning when limits are disabled
+        if (rideEntry->cars_per_flat_ride == 0xFF)
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideCreateAction.cpp:160]: Could you perhaps introduce a constant for the 0xFF value, considering it has a special meaning, and replace the existing occurrences?

**Comment by Legotown55** [comment]
When will this fix be out?

**Comment by Basssiiie** [comment]
@Legotown55 hopefully soon :)

**Review by Gymnasiast — COMMENTED** [review]
Code looks OK, needs testing.

**Comment by Basssiiie** [comment]
@Gymnasiast What kind of testing are you looking for? Manual testing by an OpenRCT2 developer or testing by just anyone else?

I've tested the change myself already of course. :)

**Comment by Gymnasiast** [comment]
Manual testing by an OpenRCT2 developer. :)

**Comment by jensj12** [comment]
> * For other rides, I did not see any relevant value in `rideEntry` that could be useful to determine a fitting proposed train amount. I set it to 12 as a kind of randomly picked value that is closer to what people would realistically pick in most cases on rides like log flumes, car rides. For rides like looping coasters it isn't that far off from a realistic number for those.

Ideally, it would pick the number it would pick without cheats, but that would require modifying the code that computes it. I looked at it when implementing the cheat, but afaik there's currently no way to know whether you're initializing the ride or checking user input (so the only way to make the cheat work was to assume user input and ignore any checks).

**Comment by oli414** [comment]
I'm testing this now, a monorail with 1 station of 5 tiles long got the suggestion for 12 trains with 8 cars per train. 

With the cheat disabled it would have suggested 1 train with 5 cars. Is there any way to keep the default amount to what it would be without the cheat enabled? Usually its an amount that works well for the ride you build, if a user needs more they can increase it themselves.

**Comment by oli414** [comment]
Since proposed_num_vehicles is a uint8, and the maximum value seems to be 32, would be possible to start using 33/255 as the value to indicate that it is unset?

**Comment by Legotown55** [comment]
When will this fix be imlomtnted

**Comment by Basssiiie** [comment]
@oli414 Without cheats the game will always try to spawn 32 trains, but it will cap at what is physically possible in terms of space on the ride. The cheat disables this cap. If we'd want to cap it at what's physically possible, then it'd require quite some refactoring. Note that the code changes are only in the 'RideCreateAction', which is only called once, as soon as you try to build 1 station piece. If you'd want to to adjust while the person builds the rest of the ride, you'd need to update that `proposed_num_vehicles` every time another track piece is built.

I'm personally not interested in doing a big amount of refactoring (I'm sorry), I just intended to make this a small fix. 

Regarding the 33/255 for `proposed_num_vehicles`, NSF allows you to have up to 255 trains, so there's unfortunately no space left in the field for an 'unset' constant (unless we change it into a `uint16_t`).

**Comment by oli414** [comment]
Fair enough, as far as the original issue goes this is still not ideal but hopefully it can be tackled in the future then. From the testing it seems to work as intended in that case.

**Pull request merged**